allennlp/modules/seq2seq_decoders/seq_decoder.py

```diff
@@ -0,0 +1,201 @@
+from typing import Dict, List, Tuple
+
+import torch
+from torch.nn import Linear
+import torch.nn.functional as F
+
+from allennlp.common.util import START_SYMBOL, END_SYMBOL
+from allennlp.data import Vocabulary
+from allennlp.models import Model
+from allennlp.modules import Embedding
+from allennlp.modules.seq2seq_decoders.decoder_cell import DecoderCell
+from allennlp.nn import util
+from allennlp.nn.beam_search import BeamSearch
+from allennlp.training.metrics import Metric, BLEU
+
+
+class SeqDecoder(Model):
```

Comment:
Why does this inherit from `Model`, instead of just `Module`?  The implicit contract of a `Model` is it's something that takes the output of some `DatasetReader` - I don't think that's what this is intended to do.

Reply:
Changed